File: 프로그래머스/1/160586. 대충 만든 자판/대충 만든 자판.swift
import Foundation

func solution(_ keymap:[String], _ targets:[String]) -> [Int] {
    var dic: [Character:Int] = [:]
    var result: [Int] = Array(repeating: 0, count: targets.count)
    
    keymap.forEach { str in
        str.enumerated().forEach { ch in
            if dic[ch.element] == nil {
                dic.updateValue(ch.offset + 1, forKey: ch.element)
            } else {
                let m = min(dic[ch.element]!, ch.offset + 1)
                dic.updateValue(m, forKey: ch.element)
            }
        }
    }
    
    for target in targets.enumerated() {
        for ch in target.element {
            if dic[ch] != nil {
                result[target.offset] += dic[ch]!
            } else {
                result[target.offset] = -1
                break
            }
        }
    }
    
    return result
}

File: 프로그래머스/1/160586. 대충 만든 자판/대충 만든 자판.cpp
#include <algorithm>
#include <iostream>
#include <string>
#include <vector>

using namespace std;

vector<int> solution(vector<string> keymap, vector<string> targets) {
    vector<int> answer;
    vector<int> key(26, 200);

    for (int i = 0; i < keymap.size(); i++) {
        for (int j = 0; j < keymap[i].size(); j++) {
            key[keymap[i][j] - 'A'] = min(key[keymap[i][j] - 'A'], j + 1);
        }
    }

    for (int i = 0; i < targets.size(); i++) {
        int result = 0;
        for (int j = 0; j < targets[i].size(); j++) {
            if(key[targets[i][j] - 'A'] == 200) {
                result = -1;
                break;
            } else {
                result += key[targets[i][j] - 'A'];
            }
        }
        answer.push_back(result);
    }

    for(auto num : answer) {
        cout << num << endl;
    }

    return answer;
}